### ISSUE TITLE
Zotero URI links result in "page not found" error

### ISSUE THREAD

**Issue by dvanoni** [dvanoni/notero#172]
**Describe the issue**
When clicking the link in a `Zotero URI` property in Notion, the user ends up on a Zotero error page with the error message: page not found.

**To Reproduce**

1. Go to a Notion page created by Notero that includes the `Zotero URI` property
2. Click on the link in the `Zotero URI` property
3. See the "page not found" error page

**Expected behavior**
Clicking the link opens the desired item in the Zotero web library.

**Screenshots**

https://user-images.githubusercontent.com/299357/202096649-17264929-c22d-432f-a228-471cd99e3e1e.mp4

**Comment by dvanoni** [comment]
I've [reached out for support in the Zotero forums](https://forums.zotero.org/discussion/101169/zotero-http-links-result-in-page-not-found).

**Comment by dvanoni** [comment]
Response from Dan Stillman (Zotero team) in the forums:

> Yes, sorry, we had to shut those off due to abuse (though note that, while `getItemURI()` worked in practice, it was never guaranteed to provide a web link — that's just a URI to identify the item).
>
> The web API provides the canonical web library URL for every item in links.alternate.href.
>
> For the client, we'll add a function to get web library links using the username slug. For now, you can use `Zotero.Users.getCurrentUsername()` and apply the [same transformation we use for the API](https://github.com/zotero/dataserver/blob/97267cbee6441350b66baff4c34e1f8f9833118a/model/Utilities.inc.php#L88-L100). (There won't be a username if the person hasn't synced.)

**Comment by sheldon123z** [comment]
Hi @dvanoni ,

my way for solving this issue is to use a chrome add-on called "URL redirect for developers" as a middle point to open the zotero application and find the item. 

the code to import into the add-on is shown below:
```
{
  "settingsSchemaVersion": 1,
  "enabled": true,
  "removeCSPRules": [],
  "redirectRules": [
    {
      "enabled": true,
      "fromPattern": "*://www.zotero.org/{user}/items/{id}",
      "toPattern": "zotero://select/library/items/{id}"
    },
    {
      "enabled": true,
      "fromPattern": "http://zotero.org/users/{user}/items/{id}",
      "toPattern": "https://www.zotero.org/{user}/items/{id}"
    }
  ]
}
```

add the extension from chrome extension market
<img width="417" alt="image" src="https://user-images.githubusercontent.com/42455304/206177446-282d3eff-059c-482f-9d61-1c7cdcc2e221.png">

click the import button and paste the codes shown above
<img width="1032" alt="image" src="https://user-images.githubusercontent.com/42455304/206177632-b3ce6a76-bbf1-4c84-9b39-6dfe36aa5c91.png">

As long as the chrome open, click on the "zotero url" the chrome will redirect back to the zotero application and find the item it points to.

**Comment by sheldon123z** [comment]
it works for me since almost all the time when I use zotero, I also keep the chrome on.

https://user-images.githubusercontent.com/42455304/206180957-add78271-0981-4c1a-a6d1-6ef7e0bea2b2.mov

**Comment by austinmw** [comment]
Also facing this problem.

**Comment by vadbars** [comment]
The problem has not been solved?

**Comment by dvanoni** [comment]
I haven't had time to get to this one yet, but it's on the todo list!

**Comment by austinmw** [comment]
By the way, happen to have any experience using LiquidText? I'd love to figure out a way to also add liquidtext links to this database. For example:

`[[GFM](lt://openProj/F76A9861-A78F-4BE3-973E-6D511348F1A7)](lt://openProj/F76A9861-A78F-4BE3-973E-6D511348F1A7)`

**Comment by dvanoni** [comment]
@austinmw, unfortunately, it looks like Notion still only supports `http` and `https` links.

You could try using the Chrome extension that @sheldon123z [mentioned above](https://github.com/dvanoni/notero/issues/172#issuecomment-1340890056) with something like the following:

**From:** `http://openProj/{projId}`
**To:** `lt://openProj/{projId}`

Then, when you paste one of those LiquidText links into Notion, you'd need to replace the leading `lt` with `http` so that Notion will recognize it as a link.

I haven't used the extension myself, so I'm not sure if I have it quite right, but let me know if you give it a try!

**Comment by dvanoni** [comment]
Another URL redirect tool was shared here: https://github.com/dvanoni/notero/issues/78#issuecomment-1378732014

**Comment by hangman1998** [comment]
Hi. thanks for this awesome tool.
I propose a little hint on README about this issue.
I started using this plugin today and was confused about why the links were not working.

**Comment by meteore** [comment]
If you're using Firefox, the extension [Firefox Redirector](https://addons.mozilla.org/en-US/firefox/addon/redirector/) provides a workaround. I had success with the following wildcard rule:

**Comment by Skywuuuu** [comment]
@sheldon123z Hi, could you please share the link to the Chrome add-on you use? I cannot find it on Chrome extension market. xiexie.

**Comment by sheldon123z** [comment]
> @sheldon123z Hi, could you please share the link to the Chrome add-on you use? I cannot find it on Chrome extension market. xiexie.

sorry reply so late, I didn't see your message, here is the extension github page, it seems like already removed from Chrome extension market. Please copy & paste the codes I provided, it should work. 

https://github.com/wavebeem/url-redirect-webextension

**Comment by sheldon123z** [comment]
@Skywuuuu 

Another point for achiving this is that keep your browser and Zotero application open at the same time, then you can click the Zotero link on notion, the browser will redirect you to Zotero and have the correct item selected.

**Comment by sheldon123z** [comment]
[url-redirect-webextension-master.zip](https://github.com/dvanoni/notero/files/12513334/url-redirect-webextension-master.zip)

**Comment by dvanoni** [comment]
This should be fixed in [v0.5.6](https://github.com/dvanoni/notero/releases/tag/v0.5.6)! Please let me know if you have any trouble with it.
